code/game/jobs/access.dm

```diff
@@ -186,6 +186,17 @@
 				return 1
 		return 0
 	return 1
+	
+/proc/has_access(var/list/req_access, var/list/req_one_access, var/list/accesses)
+	for(var/req in req_access)
+		if(!(req in accesses)) //doesn't have this access
+			return 0
+	if(req_one_access.len)
+		for(var/req in req_one_access)
+			if(req in accesses) //has an access from the single access list
+				return 1
+		return 0
+	return 1

```

Comment:
Just a minor thing, but line 180-188 above is basically replaced with a call to this proc.